include/vast/ABI/MLIRTypeInfo.hpp

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2022-present, Trail of Bits, Inc.
+
+#pragma once
+
+#include "vast/Util/Warnings.hpp"
+
+VAST_RELAX_WARNINGS
+#include <llvm/Support/Alignment.h>
+VAST_UNRELAX_WARNINGS
+
+#include "vast/ABI/ABI.hpp"
+
+#include "vast/Dialect/HighLevel/HighLevelUtils.hpp"
+
+namespace vast::abi {
+
+    struct vast_type_info {
```

Comment:
does it need to be called with prefix `vast_` as it is already in `vast::abi`

Reply:
Probably not. Will remove the prefix.